### DISCUSSION TITLE
Whitespace issues

### DISCUSSION THREAD

**Discussion by Philbywhizz** [togfoxy/MarsLander#214] [General]
Hey guys,

I'm really struggling with whitespace issues here when it comes to merging in existing or other branches into my own.

Git really throws a wobbly when there is whitespace at the end of a line as it shows up as being modified and results in merge conflicts. A good example is my current PR #199 in the functions.lua file.

It could also be the way I'm rebasing my branch on my main (I'm still working out that workflow).

Any advise on how you work around the issue (or am I the only one having the problem?)

**Comment by togfoxy** [comment]
Fair to say this time it's not my editor?

**Comment by MadByteDE** [comment]
Maybe because I removed whitespaces again in one of the #194 commits.
The problem would be solved if everyone would make sure that there are no whitespaces in the code when sending in a PR.
But I believe @togfoxy's editor might not cooperate on that task and doesn't even show whenever there are whitespaces.

Or we just ignore them and don't change them except for when we touch the code nearby. Kinda hard for me because I'm really nit picky when it comes to code clean-up (as you might know by now).

**Comment by togfoxy** [comment]
I definitely can't see them and spent a few days working out what you guys were getting made about. Can you make them invisible in your editors?

I'm thinking of trying ATOM anyway.

**Reply by MadByteDE** [reply]
> Can you make them invisible in your editors?

No we can make them VISIBLE to see where they are, or even make them bright red so that we naturally want to remove them.

Atom should be able to show them too, but I switched from Atom to VSCode because it was easier to use and had more features by default.

**Reply by Philbywhizz** [reply]
You can make them invisible, but it hides the underlying problem. Extra changes in pull requests and commits flip-flopping back and forward. It's just not good practice.
